### ISSUE TITLE
Installation does not work (for Plains Cree and non-linguistic Divvun tools)

### ISSUE THREAD

**Issue by aarppe** [divvun/divvun-installer-macos#29]
Using the new Divvun Installer (v.2.0.0), trying to install the Plains Cree spellers and keyboards as well as the non-linguistic-content Divvun tools, I get the following error, among many:

<img width="532" alt="image" src="https://user-images.githubusercontent.com/15003407/83953379-366e8a80-a7fd-11ea-903b-2a3b8fe01fba.png">

**Comment by bbqsrc** [comment]
First, give me the `sha256sum` of your package:

```
sha256sum /Library/Caches/Pahkat/packages/b5/3d/c997e6e2407ed1915b4e996c4fa857e3c775d016d3510e1c4fad55e9e4cf/speller-crk-0.1.0-nightly.20200604T135005Z.pkg
```

I get this: `9cac144ca4b7ac2816e24d48832eed1cf2f59f847b856ddfc9e180f1461f1932`

----

Then, run the following command to see if installing on the command line exhibits different behaviour:

```
sudo installer -pkg /Library/Caches/Pahkat/packages/b5/3d/c997e6e2407ed1915b4e996c4fa857e3c775d016d3510e1c4fad55e9e4cf/speller-crk-0.1.0-nightly.20200604T135005Z.pkg -target /
```

You should get some errors, reply with those errors.

If however the installation succeeds, you should see the following output:

```
installer: Package name is Plains Cree
installer: Upgrading at base path /
installer: The upgrade was successful.
installer: The install requires restarting now.
```

----

If you still have problems after reporting the output of the above, try `sudo rm -r /Library/Caches/Pahkat` and see if the problem goes away. It's always safe to delete the cache.


Debugging these installation errors are a bit of a problem with the macOS package installer is it gives that same generic error for multiple reasons:

- notarization or code signing isn't validating correctly (currently an intermittent issue on the receiving end of Catalina users)
- the cached .pkg got corrupted
- a partial download got copied into the cache directory
- potentially some issue with the bundle that makes it only compatible with newer macOS versions


Also if you could tell me what version of macOS you're running, that could help too. Thanks!

**Comment by aarppe** [comment]
1. Check-sum

```
shasum Downloads/divvun-installer_2.0.0-nightly.20200605T151551Z_macos.pkg 
b28de001868971a3309f2adb01b778e1d254e686  Downloads/divvun-installer_2.0.0-nightly.20200605T151551Z_macos.pkg
```

2. Installation

```
sudo installer -pkg Downloads/divvun-installer_2.0.0-nightly.20200605T151551Z_macos.pkg -target /
installer: Package name is Divvun Installer
installer: Upgrading at base path /
installer: The upgrade was successful.
```

Does install, but no mention of the Plains Cree tools.

My MacOS is: 10.15.3

**Comment by bbqsrc** [comment]
@aarppe you ran the commands on the Divvun Installer, not the speller-crk package. Copy and paste the commands as shown.

**Comment by bbqsrc** [comment]
Fixed the issue where corrupted downloads cause installation failures, a new build should come out in a few hours that fixes that issue.

**Comment by aarppe** [comment]
Ok, but then I get this error, i.e. no such file:

```
shasum /Library/Caches/Pahkat/packages/b5/3d/c997e6e2407ed1915b4e996c4fa857e3c775d016d3510e1c4fad55e9e4cf/speller-crk-0.1.0-nightly.20200604T135005Z.pkg
shasum: /Library/Caches/Pahkat/packages/b5/3d/c997e6e2407ed1915b4e996c4fa857e3c775d016d3510e1c4fad55e9e4cf/speller-crk-0.1.0-nightly.20200604T135005Z.pkg: No such file or directory
```

**Comment by aarppe** [comment]
```
shasum /Library/Caches/Pahkat/packages/2a/3d/1e0430c7fdae1480cbdab915236620635d8811bb7d4e4c4c817e7ca85105/keyboard-crk_0.1.1-nightly.20200605T065428Z.pkg
4c798dc62ab9b366549a0b983be5437ea8bb497a  /Library/Caches/Pahkat/packages/2a/3d/1e0430c7fdae1480cbdab915236620635d8811bb7d4e4c4c817e7ca85105/keyboard-crk_0.1.1-nightly.20200605T065428Z.pkg
```
```
sudo installer -pkg /Library/Caches/Pahkat/packages/2a/3d/1e0430c7fdae1480cbdab915236620635d8811bb7d4e4c4c817e7ca85105/keyboard-crk_0.1.1-nightly.20200605T065428Z.pkg -target / 
installer: Error - the package path specified was invalid: '/Library/Caches/Pahkat/packages/2a/3d/1e0430c7fdae1480cbdab915236620635d8811bb7d4e4c4c817e7ca85105/keyboard-crk_0.1.1-nightly.20200605T065428Z.pkg'.

```

**Comment by bbqsrc** [comment]
@aarppe: It looks like your downloads are being cut-off before they complete, which was a problem in a specific older version of the client. 

Try installing this version I freshly baked today: https://pahkat.uit.no/artifacts/divvun-installer_2.0.0-nightly.20200607T131408Z_macos.pkg - I strongly suggest rebooting after installing even if the installer doesn't suggest to do so.

Uninstallation should also now work correctly in this version. Also, you're on macOS 10.15.3 while 10.15.5 is out. I would highly recommend updating to .5, as some older versions of Catalina had spurious errors installing _any_ packages, even if they weren't from Divvun Installer, as Apple mandated this new code-signing thing called notarization, and it's not going well. :smile:

**Comment by aarppe** [comment]
Ok - now things get installed. But when I reboot, the new Divvun installer isn't showing anything. Previously, it has taken a little while before the spellers etc. become apparent, sometimes requiring a restart. I didn't get upgrade my MacOS, though.

**Comment by aarppe** [comment]
Installing the latest MacOS 10.15.5 seems to improve things.

<img width="822" alt="image" src="https://user-images.githubusercontent.com/15003407/83978307-c37c1700-a8c3-11ea-815c-c125d2fbb526.png">

<img width="710" alt="image" src="https://user-images.githubusercontent.com/15003407/83978314-d4c52380-a8c3-11ea-9216-9125341400b3.png">

**Comment by aarppe** [comment]
But now I'm getting the following error message:

<img width="532" alt="image" src="https://user-images.githubusercontent.com/15003407/83978353-22da2700-a8c4-11ea-9448-6d9dad1786ac.png">

**Comment by aarppe** [comment]
But my next question is, how do I get access to the crk-SRO spell-checker from within MS Word 2016?

**Comment by bbqsrc** [comment]
On macOS, only system spellers are installed at this time (ie, accessible in apps such as TextEdit). 

MS Office for Mac spellers is an involved process that must be manually patched on a language by language basis and is not currently available in this beta.

For this beta, LibreOffice spellers are also not available, and it is due to a bug that you can see the Divvun Tools repository at all - those are automatically installed dependencies when required and do not require user involvement and only serve to cause confusion, such as right now.

Notwithstanding, the error you saw from installing the libreoffice engine is related to Catalina not supported that pkg, as it was built before Catalina was released, and they do not show informative error messages in this case.

Anyway, as the original bug has now been fixed, I'll be closing this issue. For general questions such as the above, the appropriate venue for assistance is Zulip. thanks!

**Comment by aarppe** [comment]
Ok - but previously we had a crk spell-checker available in MacOS for Word 2016 - I am aware on the general level about how this had been implemented with patching. Also, I had a crk speller for LibreOffice as well, actually for some while. This demo is going to be a step backwards for Mac users such as myself, and I would have preferred to have known that in advance, since we are going back to what was available already a few years back.

In the above respects, this is a bug for an end-user such as myself, as the installation is not delivering what I was presuming and expecting, and which I've already had, on MacOS.

**Comment by bbqsrc** [comment]
I was mistaken about the precise status of `crk`, and as you say, it had been tested and working on macOS. Catalina also broke almost everything and we have not yet had time to set up signed builds for macOS mso spellers on our new CI/CD system yet. My statement was a general statement, and you might note that the Sami languages in the currently nightly repository also lack MSO on macOS.

MS Office spellers are however available on the Windows nightly divvun repos, so the technology can at least be demonstrated.

"This beta" means for your workshops this week. The timeline on having all these things (such as LO and MSO) functional and deployed is a matter of weeks, not months or years. LibreOffice builds being automated will also be returning. This is not a regression in technology, merely a logistical matter. These things should be prioritised amongst yourself with @snomos, as I'm merely the worker here.

I do appreciate your frustration as you have been waiting a while for these things, but this repository is for the software underlying the delivery of the content in the repositories, Divvun Installer, while your complaints are about the content of the repositories themselves. As previously stated, the best venue for those discussions would be Zulip and the audience would be @snomos.

Sorry I can't be of much more help.

**Comment by aarppe** [comment]
I understand the technical challenges - my primary gripe here is about knowing what is possible, and adjusting expectations accordingly. I've now sent out an email indicating that these tools would be available for both Win and Mac (presuming what had been possible on my own device and based on my previous discussions), and I had presumed LO as a fall-back option - had I known that it is only for Windows, I would have adjusted my message accordingly (and people might have gotten themselves a Windows machines to test out). I know that there are diehard Windows users among our co-investigators, but there are also folks who work solely with Macs.

Of course, being able to see what is possible might be entirely sufficient, and we can then get the tools installed in practice when we have the follow up sessions. We've just learnt too many times that folks are expecting tools that work on "their" gadgets "now". This is why I have been looking at whether I can get the tools installed on my Mac - I've presumed if that doesn't work then my certainty of things working for other persons is less high.

Importantly, there needs to be a concrete understanding (between @snomos and myself, and probably you as well) on what is actually being developed and made truly available for when. There was talk of us having follow-up workshops where people would try out developing spellers for their own languages and be able to get access to the results pretty much immediately through the repository integration work being done right now - there's not much point in organizing those workshops before we actually know that this will happen.
